File: app/src/main/java/com/admin/theapp/dagger/ViewModelSubComponent.java
package com.admin.theapp.dagger;

import com.admin.theapp.viewmodel.HotelDetailsViewModel;
import com.admin.theapp.viewmodel.HotelsViewModel;

import dagger.Subcomponent;

@Subcomponent
public interface ViewModelSubComponent {
    @Subcomponent.Builder
    interface Builder {
        ViewModelSubComponent build();
    }

    HotelsViewModel hotelsViewModel();

    HotelDetailsViewModel hotelDetailsViewModel();
}

File: app/src/main/java/com/admin/theapp/dagger/ViewModelSubComponent.kt
package com.admin.theapp.dagger

import com.admin.theapp.viewmodel.HotelDetailsViewModel
import com.admin.theapp.viewmodel.HotelsViewModel

import dagger.Subcomponent

@Subcomponent
interface ViewModelSubComponent {
    @Subcomponent.Builder
    interface Builder {
        fun build(): ViewModelSubComponent
    }

    fun hotelsViewModel(): HotelsViewModel

    fun hotelDetailsViewModel(): HotelDetailsViewModel
}
